src/domain/IRegistrationGateway.ts

```diff
@@ -0,0 +1,17 @@
+import { LimitedPartnership } from "@companieshouse/api-sdk-node/dist/services/limited-partnerships";
+
+import RegistrationPageType from "../presentation/controller/registration/PageType";
+
+interface IRegistrationGateway {
+  getSubmissionById(id: string): Promise<LimitedPartnership>;
+  createTransaction(
+    registrationPageType: RegistrationPageType
+  ): Promise<string>;
+  createSubmission(
```

Comment:
I was just wondering whether transactions and submissions will need to be separated rather than in one place. As long as this remains the same and does not need to be expanded to include updates and deletions it should be find. However, in the case where there is more than one function for submissions would it become a separate concern?

Reply:
These are two different requests which may succeed or fail for different reasons
one is made to the service that deals with transitions
the other to the api

Reply:
Exactly, so I was asking whether they should be separated.

Reply:
because they are different!